packages/api/src/beacon/routes/proof.ts

```diff
@@ -8,8 +8,7 @@ import {ArrayOf} from "../../utils/codecs.js";
 import {VersionCodec, VersionMeta} from "../../utils/metadata.js";
 
 export const CompactMultiProofType = new ContainerType({
-  // TODO ensure limit of all lists is sane
-  leaves: ArrayOf(ssz.Root),
+  leaves: ArrayOf(ssz.Root, 10000),
```

Comment:
I am summing this is sufficient and sane enough, same value we use in
https://github.com/ChainSafe/lodestar/blob/092be931f4d71c5f012eea726a73dab287e14306/packages/api/src/beacon/routes/lightclient.ts#L20

but I am not that familiar with return values of proof routes, would be good to get eyes on this. Better to have a too high limit, than breaking large proofs by setting it too low.

Reply:
10k is plenty big, I wanna say we had 512 set as MAX_PROOF_GINDICES at some point.

Reply:
we have a configurable limit on the server which as you noted is 512. the limit in the ssz type might still be useful to avoid crashing the client

https://github.com/ChainSafe/lodestar/blob/f21fc3c6ab567f4ddf2cf7b33dab48c7c835bed4/packages/beacon-node/src/api/impl/proof/index.ts#L13-L15

preventing server dos vector should be covered